amethyst_assets/src/simple.rs

```diff
@@ -26,19 +32,83 @@ use {Asset, AssetSpec, Cache, Context};
 /// duplicated buffer allocations, make sure your handle is reference-counted,
 /// so wrap it with an `Arc` in case the handle doesn't have this functionality
 /// itself.
+
+struct AssetUpdate<A> {
+    counter: AtomicUsize,
+    ready: RwLock<Option<A>>,
+    defer: RwLock<Option<AssetFuture<AssetPtr<A>>>>,
+}
+
+impl<A> AssetUpdate<A>
+    where A: Clone
+{
+    fn new() -> Self {
+        AssetUpdate {
+            counter: AtomicUsize::new(0),
+            ready: RwLock::new(None),
+            defer: RwLock::new(None),
+        }
+    }
+
+    fn push_update(&self, mut updated: AssetFuture<AssetPtr<A>>) {
+        match updated.poll() {
+            Ok(Async::Ready(updated)) => {
+                *self.ready.write() = Some(updated.inner.clone());
+                self.counter.fetch_add(1, Ordering::Release);
+            }
+            Err(err) => {},
```

Comment:
`Err(err)` should be `Err(_)` to silence the compiler warning, as `err` is unused